prov/efa/src/efa_cq.c

```diff
@@ -331,11 +340,119 @@ int efa_cq_open(struct fid_domain *domain_fid, struct fi_cq_attr *attr,
 err_destroy_pool:
 	ofi_bufpool_destroy(cq->wce_pool);
 err_destroy_cq:
-	ibv_destroy_cq(cq->ibv_cq);
+	ibv_destroy_cq(ibv_cq_ex_to_cq(cq->ibv_cq_ex));
 err_free_util_cq:
 	ofi_cq_cleanup(&cq->util_cq);
 err_free_cq:
 	free(cq);
 	return ret;
 }
 
+
+#if EFA_UNIT_TEST
+
+#include "efa_unit_tests.h"
+
+struct readerr_input {
+	struct fi_cq_err_entry *entry;
+	struct efa_cq *efa_cq;
+};
+
+static enum ibv_wc_opcode _read_send_code(struct ibv_cq_ex *current)
+{
+	return IBV_WC_SEND;
+};
+
+static int setup_readerr(void **state)
```

Comment:
Mock out the objects directly. This eliminates dependencies on other function/API, and more importantly does not require EFA device to run the tests.